newm-server/src/test/kotlin/io/newm/server/features/arweave/repo/UtilTest.kt

```diff
@@ -0,0 +1,164 @@
+                    arweaveClipUrl = "ar://QpgjmWmAHNeRVgx_Ylwvh16i3aWd8BBgyq7f16gaUu0",
+                    album = "Daisuke",
+                    duration = 200000,
+                    track = 1,
+                    compositionCopyrightOwner = "Mirai Music Publishing",
+                    compositionCopyrightYear = 2023,
+                    phonographicCopyrightOwner = "Danketsu, Mirai Music, NSTASIA",
+                    phonographicCopyrightYear = 2023,
+                    mintingStatus = MintingStatus.Pending
+                )
+
+            actual = Util.weaveRequest(song)
+
+            val json =
+                Json {
+                    ignoreUnknownKeys = true
+                    explicitNulls = false
+                    isLenient = true
+                }
+
+            val expected =
+                WeaveRequest(
+                    json.encodeToString(
+                        WeaveProps(
+                            arweaveWalletJson = "",
+                            files = emptyList(),
+                            checkAndFund = false
+                        )
+                    )
+                )
+
+            assertThat(actual).isEqualTo(expected)
+        }
+
+    @Test
+    fun `test weaveRequest doesn't exist`() =
+        runTest {
+            var actual: WeaveRequest
+
+            val song =
+                Song(
+                    ownerId = UUID.randomUUID(),
+                    title = "Daisuke",
+                    genres = listOf("Pop", "House", "Tribal"),
+                    releaseDate = LocalDate.parse("2023-02-03"),
+                    publicationDate = LocalDate.parse("2023-02-03"),
+                    isrc = "QZ-NW7-23-57511",
+                    moods = listOf("spiritual"),
+                    arweaveCoverArtUrl = "ar://GlMlqHIPjwUtlPUfQxDdX1jWSjlKK1BCTBIekXgA66A",
+                    arweaveLyricsUrl = "ar://7vQTHTkgybn8nVLDlukGiBazy2NZVhWP6HZdJdmPH00",
+                    arweaveClipUrl = "ar://QpgjmWmAHNeRVgx_Ylwvh16i3aWd8BBgyq7f16gaUu0",
+                    tokenAgreementUrl = "https://newm.io/agreement",
+                    album = "Daisuke",
+                    duration = 200000,
+                    track = 1,
+                    compositionCopyrightOwner = "Mirai Music Publishing",
+                    compositionCopyrightYear = 2023,
+                    phonographicCopyrightOwner = "Danketsu, Mirai Music, NSTASIA",
+                    phonographicCopyrightYear = 2023,
+                    mintingStatus = MintingStatus.Pending
+                )
+
+            actual = Util.weaveRequest(song)
+
+            val json =
+                Json {
+                    ignoreUnknownKeys = true
+                    explicitNulls = false
+                    isLenient = true
+                }
+
+            val expected =
+                WeaveRequest(
+                    json.encodeToString(
```

Comment:
Since what you are testing is the json encoding itself, don't encode the json in your expected value. Just put a hardcoded string in here of what you expect. That way, if the runtime code changes, we will see a difference here and the test will fail.